test/e2e/datafederation_pe_test.go

```diff
@@ -127,5 +130,25 @@ var _ = Describe("UserLogin", Label("datafederation"), func() {
 			}, df)).To(Succeed())
 			Expect(testData.K8SClient.Delete(testData.Context, df)).Should(Succeed())
 		})
+
+		By("Deleting DataFederation Private Endpoint", func() {
+			// This is required or will result on error:
+			// CANNOT_CLOSE_GROUP_ACTIVE_ATLAS_DATA_FEDERATION_PRIVATE_ENDPOINTS
+			// for some reason, doing it just once doesn't work
+			// TODO: revisit and cleanup once CLOUDP-280905 is fixed
+			for i := 0; i < 5; i++ { // 3 retries or less seems to fail
+				_, resp, err := atlasClient.Client.DataFederationApi.DeleteDataFederationPrivateEndpoint(testData.Context,
+					testData.Project.ID(), pe.ID).Execute()
+				if admin.IsErrorCode(err, "RESOURCE_NOT_FOUND") {
+					msg := fmt.Sprintf("It says it Removed DataFederation Private Endpoint %q for project %s after retry %d\n",
+						pe.ID, testData.Project.ID(), i+1)
+					GinkgoWriter.Write([]byte(msg))
+					continue
+				}
+				Expect(err).To(BeNil())
+				Expect(resp).NotTo(BeNil())
+				Expect(resp.StatusCode).To(BeEquivalentTo(http.StatusNoContent))
+			}
```

Comment:
I'd do Ginkgo style

```suggestion
			Eventually(func(g Gomega) {
				_, resp, err := atlasClient.Client.DataFederationApi.
					DeleteDataFederationPrivateEndpoint(testData.Context, testData.Project.ID(), pe.ID).
					Execute()

				g.Expect(err).To(BeNil(), fmt.Sprintf("deletion of private endpoint failed with error %s", err))
				g.Expect(resp).NotTo(BeNil())
				g.Expect(resp.StatusCode).To(BeEquivalentTo(http.StatusNoContent))
			}).WithTimeout(5 * time.Minute).WithPolling(15 * time.Second).Should(Succeed())
```

Reply:
This would not work.

I can actually use an `Eventually`, as you and @igor-karpukhin suggested, but I need to expect at the return to be successful twice. Once is not enough due to the server side issue.

I will rewrite, but it will be a bit different than what is suggested here.